projects/ngx-sub-form/src/lib/new/helpers.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  AbstractControlOptions,
+  ControlValueAccessor,
+  FormArray,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+} from '@angular/forms';
+import { ReplaySubject } from 'rxjs';
+import { Nilable } from 'tsdef';
+import {
+  ArrayPropertyKey,
+  ControlsNames,
+  NewFormErrors,
+  OneOfControlsTypes,
+  TypedFormGroup,
+} from '../ngx-sub-form-utils';
+import {
+  ControlValueAccessorComponentInstance,
+  FormBindings,
+  NgxSubFormArrayOptions,
+  NgxSubFormOptions,
+} from './ngx-sub-form.types';
+
+export const deepCopy = <T>(value: T): T => JSON.parse(JSON.stringify(value));
+
+export const patchClassInstance = (componentInstance: any, obj: Object) => {
+  Object.entries(obj).forEach(([key, newMethod]) => {
+    componentInstance[key] = newMethod;
```

Comment:
What happens if the instance already has one of the hooks defined? is it on purpose we do not use `safelyPatchClassInstance` ?